contracts/ERC677MultiBridgeToken.sol

```diff
@@ -0,0 +1,108 @@
+        bridgePointers[F_ADDR] = F_ADDR; // empty bridge contracts list
+    }
+
+    /**
+    * @dev Adds one more bridge contract into the list
+    * @param _bridge bridge contract address
+    */
+    function setBridge(address _bridge) external onlyOwner {
+        require(bridgeCount < MAX_BRIDGES);
+        require(AddressUtils.isContract(_bridge));
+        require(!isBridge(_bridge));
+
+        address firstBridge = bridgePointers[F_ADDR];
+        require(firstBridge != address(0));
+        bridgePointers[F_ADDR] = _bridge;
+        bridgePointers[_bridge] = firstBridge;
+        bridgeCount = bridgeCount.add(1);
+
+        emit BridgeAdded(_bridge);
+    }
+
+    /**
+    * @dev Removes one existing bridge contract from the list
+    * @param _bridge bridge contract address
+    */
+    function unsetBridge(address _bridge) external onlyOwner {
+        require(isBridge(_bridge));
+
+        address nextBridge = bridgePointers[_bridge];
+        address index = F_ADDR;
+        address next = bridgePointers[index];
+        require(next != address(0));
+
+        while (next != _bridge) {
+            index = next;
+            next = bridgePointers[index];
+
+            require(next != F_ADDR && next != address(0));
+        }
+
+        bridgePointers[index] = nextBridge;
+        delete bridgePointers[_bridge];
+        bridgeCount = bridgeCount.sub(1);
+
+        emit BridgeRemoved(_bridge);
+    }
+
+    /**
+    * @dev Returns all recorded bridge contract addresses
+    * @return address[] bridge contract addresses
+    */
+    function bridgeList() external view returns (address[]) {
+        address[] memory list = new address[](bridgeCount);
+        uint256 counter = 0;
+        address nextBridge = bridgePointers[F_ADDR];
+        require(nextBridge != address(0));
+
+        while (nextBridge != F_ADDR) {
+            list[counter] = nextBridge;
+            nextBridge = bridgePointers[nextBridge];
+            counter++;
+
+            require(nextBridge != address(0));
+        }
+
+        return list;
+    }
+
+    /**
+    * @dev Checks if given address is included into bridge contracts list
+    * @param _address bridge contract address
+    * @return bool true, if given address is a known bridge contract 
+    */
+    function isBridge(address _address) public view returns (bool) {
+        return _address != F_ADDR && bridgePointers[_address] != address(0);
+    }
+
+    /**
+    * @dev Overrides super.callAfterTransfer, uses custom checking if an address is a known bridge contract
+    * @param _from tokens sender address
+    * @param _to tokens receiver address
+    * @param _value amount of transferred tokens
+    */
+    function callAfterTransfer(address _from, address _to, uint256 _value) internal {
+        if (AddressUtils.isContract(_to) && !contractFallback(_from, _to, _value, new bytes(0))) {
+            require(!isBridge(_to));
```

Comment:
So, the difference is only this line. Does it make sense to implement something like this:

_ERC677BridgeToken_
```solidity
function callAfterTransfer(address _from, address _to, uint256 _value) internal {
    if (AddressUtils.isContract(_to) && !contractFallback(_from, _to, _value, new bytes(0))) {
        require(!isBridge(_to));
        emit ContractFallbackCallFailed(_from, _to, _value);
    }
}

function isBridge(address _address) public view returns (bool) {
    return _address == bridgeContract;
}
```
In this case in _ERC677MultiBridgeToken_ you will need to define only `isBridge` without necessity to duplicate `callAfterTransfer`.

What do you think?